src/seedu/addressbook/Main.java

```diff
@@ -105,16 +107,21 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException soe) {
+            ui.showToUser(MESSAGE_READ_ONLY);
+            start(new String[0]);
```

Comment:
This is one of the ways you can handle the exception. 
Think how you can improve the user experience in light of the exception that just occurred, and not lose the data.